deps/curl-8.1.2/lib/vtls/sf_ocsp.c

```diff
@@ -753,6 +763,21 @@ static OCSP_RESPONSE * queryResponderUsingCurl(char *url, OCSP_CERTID *certid, c
 
   infof(data, "OCSP fetch URL: %s", urlbuf);
 
+  int url_parse_result;
+
+  url_parse_result = OCSP_parse_url(
+      urlbuf, &host, &port, &path, &use_ssl);
+  if (!url_parse_result)
+  {
+    failf(data, "Invalid OCSP fetch URL: %s", urlbuf);
+    goto end;
+  }
+  if (strcasecmp(host, last_timeout_host) == 0)
+  {
+    // skip if we got timeout on the same host for previous certificate entry
+    goto end;
```

Comment:
nit: if we avoid goto.

Reply:
goto is everywhere in the existing OCSP implementation. Leave it for now.